cmd/mapcidr/main.go

```diff
@@ -374,12 +387,33 @@ func process(wg *sync.WaitGroup, chancidr, outputchan chan string) {
 
 	// Aggregate all ips into the minimal subset possible
 	if options.Aggregate {
-		cCidrsIPV4, cCidrsIPV6 := mapcidr.CoalesceCIDRs(allCidrs)
-		for _, cidrIPV4 := range cCidrsIPV4 {
-			outputchan <- cidrIPV4.String()
-		}
-		for _, cidrIPV6 := range cCidrsIPV6 {
-			outputchan <- cidrIPV6.String()
+		// Find out the possible CIDR prefixes from given IP address range
+		if options.CIDRFromIPRange {
+			// check if ipRange has more than 2 values
+			if len(ipRange)%2 == 1 {
+				gologger.Fatal().Msgf("IP range can not have more than 2 values.")
+			}
+			// check if range is valid or not
+			if bytes.Compare(ipRange[0], ipRange[1]) > 0 {
+				gologger.Fatal().Msgf("Start IP:%s must be less than End IP:%s", ipRange[0], ipRange[1])
+			}
+			cidrs := mapcidr.RangeToCIDRs(ipRange[0], ipRange[1])
+			sort.Slice(cidrs, func(i, j int) bool {
+				return bytes.Compare(cidrs[i].IP, cidrs[j].IP) < 0
+			})
+			for _, cidr := range cidrs {
+				outputchan <- cidr.String()
+			}
+			options.CIDRFromIPRange = false
```

Comment:
What is the purpose of setting this to false?

Reply:
I forgot to remove it. It was just for testing.